assets/js/type-defs/registered-payment-method-props.js

```diff
@@ -0,0 +1,206 @@
+ *                                                       shipping.
+ */
+
+/**
+ * @typedef BillingDataProps
+ *
+ * @property {CartBillingAddress} billingData    The address used for billing.
+ * @property {Function}           setBillingData Used to set the cart billing
+ *                                               address.
+ * @property {Object}             order          The order object for the
+ *                                               purchase.
+ * @property {boolean}            orderLoading   True if the order is being
+ *                                               loaded.
+ * @property {CartTotalItem}      cartTotal      The total item for the cart.
+ * @property {SiteCurrency}       currency       Currency object.
+ * @property {string}             country        ISO country code for the
+ *                                               default country for the site.
+ * @property {CartTotalItem[]}    cartItems      The various subtotal amounts.
+ * @property {string[]}           appliedCoupons All the coupons that were
+ *                                               applied.
+ */
+
+/**
+ * @typedef EventRegistrationProps
+ *
+ * @property {function()} onCheckoutCompleteSuccess Used to subscribe callbacks
+ *                                              firing when checkout has
+ *                                              completed processing
+ *                                              successfully.
+ * @property {function()} onCheckoutCompleteError Used to subscribe callbacks
+ *                                              firing when checkout has
+ *                                              completed processing with an
+ *                                              error.
+ * @property {function()} onCheckoutProcessing  Used to subscribe callbacks
+ *                                              that will fire when checkout
+ *                                              begins processing (as a part
+ *                                              of the processing process).
+ * @property {function()} onShippingRateSuccess Used to subscribe callbacks
+ *                                              that will fire when shipping
+ *                                              rates for a given address have
+ *                                              been received successfully.
+ * @property {function()} onShippingRateFail    Used to subscribe callbacks
+ *                                              that will fire when retrieving
+ *                                              shipping rates failed.
+ * @property {function()} onShippingRateSelectSuccess Used to subscribe
+ *                                              callbacks that will fire after
+ *                                              selecting a shipping rate
+ *                                              successfully.
+ * @property {function()} onShippingRateSelectFail Used to subscribe callbacks
```

Comment:
Could probably factor out the "used to subscribe callbacks firing when" here into a common doc that applies to all :)

Reply:
Sorry, not sure I follow what you mean here? Each of these observer registration functions are for something different, so while they all are for subscribing observers to an event (the common bit of the doc), they are for specific events. 

The doc blocks are useful for getting IDE tips when hovering over an object given this named type - moving it somewhere else and referencing it would lose that description for a property.

It's entirely possible I'm missing your intent by this comment though and you had in mind something that doesn't affect the above?